modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/N4jsClientCodegen.java

```diff
@@ -0,0 +1,668 @@
+import static org.openapitools.codegen.CodegenConstants.API_PACKAGE_DESC;
+import static org.openapitools.codegen.CodegenConstants.MODEL_PACKAGE;
+import static org.openapitools.codegen.CodegenConstants.MODEL_PACKAGE_DESC;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.text.StringEscapeUtils;
+import org.openapitools.codegen.CliOption;
+import org.openapitools.codegen.CodegenConfig;
+import org.openapitools.codegen.CodegenModel;
+import org.openapitools.codegen.CodegenOperation;
+import org.openapitools.codegen.CodegenProperty;
+import org.openapitools.codegen.CodegenResponse;
+import org.openapitools.codegen.CodegenSecurity;
+import org.openapitools.codegen.CodegenType;
+import org.openapitools.codegen.DefaultCodegen;
+import org.openapitools.codegen.IJsonSchemaValidationProperties;
+import org.openapitools.codegen.SupportingFile;
+import org.openapitools.codegen.config.GlobalSettings;
+import org.openapitools.codegen.model.ModelMap;
+import org.openapitools.codegen.model.ModelsMap;
+import org.openapitools.codegen.model.OperationMap;
+import org.openapitools.codegen.model.OperationsMap;
+import org.openapitools.codegen.utils.ModelUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.media.ArraySchema;
+import io.swagger.v3.oas.models.media.ComposedSchema;
+import io.swagger.v3.oas.models.media.Schema;
+import io.swagger.v3.oas.models.parameters.Parameter;
+
+public class N4jsClientCodegen extends DefaultCodegen implements CodegenConfig {
+    public static final String CHECK_REQUIRED_PARAMS_NOT_NULL = "checkRequiredParamsNotNull";
+    public static final String CHECK_SUPERFLUOUS_BODY_PROPS = "checkSuperfluousBodyProps";
+    public static final String GENERATE_DEFAULT_API_EXECUTER = "generateDefaultApiExecuter";
+
+    final Logger LOGGER = LoggerFactory.getLogger(N4jsClientCodegen.class);
+
+    final Set<String> forbiddenChars = new HashSet<>();
+
+    private boolean checkRequiredBodyPropsNotNull = true;
+    private boolean checkSuperfluousBodyProps = true;
+    private boolean generateDefaultApiExecuter = true;
+
+    public CodegenType getTag() {
+        return CodegenType.CLIENT;
+    }
+
+    public String getName() {
+        return "n4js";
+    }
+
+    public String getHelp() {
+        return "Generates a n4js client.";
+    }
+
+    public N4jsClientCodegen() {
+        super();
+
+        // disable since otherwise Modules/Class are not generated iff used as
+        // parameters only
+        GlobalSettings.setProperty("skipFormModel", "false");
```

Comment:
can you move this line to processOpts instead?

this force other generators to have the same setting as it's a global property